homeassistant/components/home_connect/entity.py

```diff
@@ -19,37 +20,52 @@ class HomeConnectEntity(Entity):
     _attr_should_poll = False
     _attr_has_entity_name = True
 
-    def __init__(self, device: HomeConnectDevice, desc: EntityDescription) -> None:
+    @staticmethod
+    def create_unique_id(appliance: HomeAppliance, key: str) -> str:
+        """Create unique id for entity."""
+        return f"{appliance.ha_id}-{key}"
+
+    def __init__(
+        self,
+        coordinator: HomeConnectCoordinator,
+        appliance: HomeConnectApplianceData,
+        desc: EntityDescription,
+    ) -> None:
         """Initialize the entity."""
-        self.device = device
+        self.coordinator = coordinator
+        self.appliance = appliance
         self.entity_description = desc
-        self._attr_unique_id = f"{device.appliance.haId}-{self.bsh_key}"
+        self._attr_unique_id = HomeConnectEntity.create_unique_id(
+            appliance.info, desc.key
+        )
         self._attr_device_info = DeviceInfo(
-            identifiers={(DOMAIN, device.appliance.haId)},
-            manufacturer=device.appliance.brand,
-            model=device.appliance.vib,
-            name=device.appliance.name,
+            identifiers={(DOMAIN, appliance.info.ha_id)},
+            manufacturer=appliance.info.brand,
+            model=appliance.info.vib,
+            name=appliance.info.name,
         )
 
     async def async_added_to_hass(self) -> None:
-        """Register callbacks."""
-        self.async_on_remove(
-            async_dispatcher_connect(
-                self.hass, SIGNAL_UPDATE_ENTITIES, self._update_callback
-            )
+        """Register listener."""
+        await super().async_added_to_hass()
+        self.coordinator.add_home_appliances_event_listener(
+            self.appliance.info.ha_id,
+            EventKey(self.bsh_key),
+            self._async_event_update_listener,
+        )
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Unregister listener."""
+        await super().async_will_remove_from_hass()
+        self.coordinator.delete_home_appliances_event_listener(
+            self.appliance.info.ha_id,
+            EventKey(self.bsh_key),
+            self._async_event_update_listener,
         )
 
-    @callback
-    def _update_callback(self, ha_id: str) -> None:
-        """Update data."""
-        if ha_id == self.device.appliance.haId:
-            self.async_entity_update()
-
-    @callback
-    def async_entity_update(self) -> None:
-        """Update the entity."""
-        _LOGGER.debug("Entity update triggered on %s", self)
-        self.async_schedule_update_ha_state(True)
+    @abstractmethod
+    async def _async_event_update_listener(self, event: Event) -> None:
```

Comment:
Inherit from `CoordinatorEntity` instead. It will have a method that is called by the coordinator on update.